### DISCUSSION TITLE
File name display - Strip .gcode

### DISCUSSION THREAD

**Discussion by ledcomp** [Jyers/Marlin#550] [General Discussion]
Is there a way to strip the ".gcode" from the file name on the top line of the display? 
It would prevent the scrolling of some file names.

Thanks

**Comment by aryajuanda** [comment]
For the octoprint printing i use this on my start gcode :

M75 ;Start The Print Job
M117 <F>{{ event.name[:-6] }} ;Sent Filename to Display

**Reply by ledcomp** [reply]
I tried this in Cura but unfortunately it did not work.

**Reply by LABMONITOR** [reply]
```[:-6]```  is shorthand for Python string truncation. Additionally, the object it is used with as well is a call on Octoprint's API, -- Cura will not recognize this since this M117 command is processed by Octoprint before it sends it to the printer.

Stripping it this way, however, doesn't actually remove '.gcode' on the actual raw file name (and thus character count), but only specifically on that M117 message. With how the FW currently handles this it will still scroll if it reaches the character count threshold. But I see that @Jyers may be implementing an option to strip this so I am sure that within his good efforts this wouldn't be an issue.

**Comment by Jyers** [comment]
This should be doable. I'll put it on my todo list.

**Reply by ledcomp** [reply]
Wow, that was easy, Thanks
